log/pom.xml

```diff
@@ -49,6 +49,30 @@
       <groupId>org.osgi</groupId>
       <artifactId>osgi.core</artifactId>
       <version>6.0.0</version>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.osgi</groupId>
+      <artifactId>org.osgi.namespace.service</artifactId>
+      <version>1.0.0</version>
+    </dependency>
+    <dependency>
+      <groupId>org.osgi</groupId>
+      <artifactId>osgi.annotation</artifactId>
+      <version>8.1.0</version>
+      <optional>true</optional>
+    </dependency>
+    <dependency>
+      <groupId>org.osgi</groupId>
+      <artifactId>org.osgi.service.component.annotations</artifactId>
+      <version>1.5.1</version>
+      <optional>true</optional>
```

Comment:
It makes it cleaner for users of this artifact. The annotations are only needed during compile time and are then discarded. The `provided` scope shows up in project dependencies for anyone who would use this as a dependency. While `optional` dependencies do not propagate automatically and have to be added by user of this artifact manually. See "[How do optional dependencies work?](https://maven.apache.org/guides/introduction/introduction-to-optional-and-excludes-dependencies.html)" in Maven documentation.

Reply:
I don't think that optional is correct here. As you mentioned the doc states that optional should be used if there are different providers (e.g. slf4j, jpa, jms,...) and not to hide something (what is shown to the user is then more a decision of the tooling)

Reply:
You should be able to use `<scope>provided</scope>` for this case.

Reply:
> You should be able to use `<scope>provided</scope>` for this case.

I have removed the `optional`. I don't think `provided` scope is appropriate either because there is no other provider of these. Nothing bad will happen if the artifacts stay on classpath.

Reply:
The annotations are just relevant on build time for tools loke bnd. If the jar is build the annotation is not visuable at runtime.

Thats why provided is best Option